src/core/geometry/qgsgeometryeditutils.cpp

```diff
@@ -26,11 +26,12 @@ email                : marco.hugentobler at sourcepole dot com
 #include "qgsvectorlayer.h"
 #include <limits>
 
-int QgsGeometryEditUtils::addRing( QgsAbstractGeometry *geom, QgsCurve *ring )
+QgsGeometry::OperationResult QgsGeometryEditUtils::addRing( QgsAbstractGeometry *geom, QgsCurve *r )
 {
+  std::unique_ptr<QgsCurve> ring( r );
```

Comment:
I think that's what's done here. If the `unique_ptr` scopes out, `r` is deleted.